ui/pages/login.tsx

```diff
@@ -245,7 +312,33 @@ const Login: NextPage = () => {
   const supportsWebauthn = flow?.ui.nodes.some(
     (node) => node.group === "webauthn",
   );
-  const renderFlow = isAuthCode ? filterFlow(replaceAuthLabel(flow)) : flow;
+
+  const renderFlow: LoginFlow | undefined = flow
+    ? isIdentifierFirst
+      ? {
+          ...flow,
+          ui: {
+            ...flow.ui,
+            nodes: flow.ui.nodes.filter((n: UiNode) => {
+              if (
+                n.attributes.node_type === "input" &&
+                typeof (n.attributes as UiNodeInputAttributes).name === "string"
+              ) {
+                const name = (n.attributes as UiNodeInputAttributes).name;
+                return (
+                  name === "identifier" ||
+                  name === "csrf_token" ||
+                  name === "method"
+                );
+              }
+              return false;
+            }),
+          },
+        }
+      : isAuthCode
+        ? filterFlow(replaceAuthLabel(flow))
+        : flow
```

Comment:
issue: nested ternary operator inside ternary operator == hurts to read
we need to improve readability of the code, can you please break it down?